packages/playground/blueprints/src/lib/steps/run-sql.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { NodePHP } from '@php-wasm/node';
+import { compileBlueprint, runBlueprintSteps } from '../compile';
+import { phpVars } from '@php-wasm/util';
+
+const phpVersion = '8.0';
+describe('Blueprint step runSql', () => {
+	let php: NodePHP;
+
+	beforeEach(async () => {
+		php = await NodePHP.load(phpVersion, {
+			requestHandler: {
+				documentRoot: '/wordpress',
+			},
+		});
+	});
+
+	it('should split and "run" sql queries', async () => {
+		const encoder = new TextEncoder();
+		const docroot = '/wordpress';
+		const sqlFilename = `/tmp/${crypto.randomUUID()}.sql`;
+		const resFilename = `/tmp/${crypto.randomUUID()}.json`;
+		const js = phpVars({ docroot, sqlFilename, resFilename });
+		await php.mkdir(docroot);
+
+		// Create an object that will log all function calls
+		await php.writeFile(
+			`${docroot}/wp-load.php`,
+			`<?php
+			class MockLogger
+			{
+				public function __call($function, $args)
+				{
+					$entry = (object)[
+						'type' => 'CALL',
+						'function' => $function,
+						'args' => $args,
+					];
+
+					file_put_contents(${js.resFilename}, json_encode($entry) . "\n", FILE_APPEND);
+				}
+			}
+
+			global $wpdb;
+			$wpdb = new MockLogger();
+			file_put_contents(${js.resFilename}, '');
+			`
+		);
+
+		// Test a single query
+		await runBlueprintSteps(
+			compileBlueprint({
+				steps: [
+					{
+						step: 'runSql',
+						sql: {
+							resource: 'literal',
+							contents: encoder.encode(`SELECT * FROM wp_users;`),
+							name: 'single-query.sql',
+						},
+					},
+				],
+			}),
+			php
+		);
```

Comment:
Almost there! The `encoder.encode()` shouldn't be required, it seems like a bug in the Node.js unit test setup. You could just call your step function directly in here instead of going through `compileBlueprint`:

```suggestion
		const mockFile = {
			name: 'queries.sql',
			async arrayBuffer() {
				return new TextEncoder().encode('SELECT * FROM wp_users;')
					.buffer;
			},
			type: 'text/plain',
		} as any;
		await runSql( php, { 
			sql: mockFile
		});
```

Ideally mocking a File instance like that wouldn't be needed. We need that, because the one provided by Node/Vitest has no `arrayBuffer()` method. Once the playground-tools work progresses, we'll be able to lean on a polyfill and just do `new File()`

Reply:
@adamziel I made the suggested change.